flink-runtime/src/test/java/org/apache/flink/runtime/rpc/RpcEndpointTest.java

```diff
@@ -340,108 +319,54 @@ public void testExecute() throws InterruptedException, ExecutionException, Timeo
-        final CompletableFuture<Long> actualDelayMsFuture2 = new CompletableFuture<>();
-        final RpcEndpoint endpoint = new BaseEndpoint(rpcService);
-        final int expectedInt = 12345;
-        final String expectedString = "Flink";
-        try {
-            endpoint.start();
-            final long startTime = System.currentTimeMillis();
-            final ScheduledFuture<Integer> intScheduleFuture =
-                    endpoint.getMainThreadExecutor()
-                            .schedule(
-                                    () -> {
-                                        endpoint.validateRunsInMainThread();
-                                        actualDelayMsFuture1.complete(
-                                                System.currentTimeMillis() - startTime);
-                                        return expectedInt;
-                                    },
-                                    expectedDelay1.getSize(),
-                                    expectedDelay1.getUnit());
-            final ScheduledFuture<String> stringScheduledFuture =
-                    endpoint.getMainThreadExecutor()
-                            .schedule(
-                                    () -> {
-                                        endpoint.validateRunsInMainThread();
-                                        actualDelayMsFuture2.complete(
-                                                System.currentTimeMillis() - startTime);
-                                        return expectedString;
-                                    },
-                                    expectedDelay2.getSize(),
-                                    expectedDelay2.getUnit());
-
-            final long actualDelayMs1 =
-                    actualDelayMsFuture1.get(
-                            expectedDelay1.getSize() * 2, expectedDelay1.getUnit());
-            final long actualDelayMs2 =
-                    actualDelayMsFuture2.get(
-                            expectedDelay2.getSize() * 2, expectedDelay2.getUnit());
-            final int actualInteger =
-                    intScheduleFuture.get(expectedDelay1.getSize() * 2, expectedDelay1.getUnit());
-            final String actualString =
-                    stringScheduledFuture.get(
-                            expectedDelay2.getSize() * 2, expectedDelay2.getUnit());
-            assertTrue(actualDelayMs1 > expectedDelay1.toMilliseconds() * 0.8);
-            assertTrue(actualDelayMs1 < expectedDelay1.toMilliseconds() * 1.2);
-            assertTrue(actualDelayMs2 > expectedDelay2.toMilliseconds() * 0.8);
-            assertTrue(actualDelayMs2 < expectedDelay2.toMilliseconds() * 1.2);
-            assertEquals(expectedInt, actualInteger);
-            assertEquals(expectedString, actualString);
-        } finally {
-            RpcUtils.terminateRpcEndpoint(endpoint, TIMEOUT);
-        }
+    public void testScheduleRunnableWithDelayInSeconds() throws Exception {
```

Comment:
Not sure whether we really need the variants where we test the delay in seconds. I would be fine with removing them.

Reply:
The original issue, for which the tests where added, was that different time units were not handled correctly. I'm not really happy with testing _that_, but in this particular case it seems appropriate, purely due to precedence.

Reply:
(That said, the test wasn't actually guarding against the original issue...)

Reply:
Ah ok, so we are testing the conversion in `RpcEndpoint.MainThreadExecutor` here. This makes sense then.